pySDC/core/Sweeper.py

```diff
@@ -270,40 +280,96 @@ def rho(x):
             self.parallelizable = True
 
         elif qd_type == "MIN-SR-S":
+
             M = QDmat.shape[0] - 1
-            Q = coll.Qmat[1:, 1:]
-            nodes = coll.nodes
+            quadType = coll.quad_type
+            nodeType = coll.node_type
+
+            # Main function to compute coefficients
+            def computeCoeffs(m, a=None, b=None):
+                coll = CollBase(num_nodes=m, node_type=nodeType, quad_type=quadType)
+
+                Q = coll.Qmat[1:, 1:]
+                nodes = coll.nodes
+
+                nCoeffs = len(nodes)
+                if quadType in ['LOBATTO', 'RADAU-LEFT']:
+                    nCoeffs -= 1
+                    Q = Q[1:, 1:]
+                    nodes = nodes[1:]
+
+                if nCoeffs == 1:
+                    coeffs = np.diag(Q)
+
+                else:
+
+                    def nilpotency(coeffs):
+                        """Function verifying the nilpotency from given coefficients"""
+                        coeffs = np.asarray(coeffs)
+                        kMats = [(1 - z) * np.eye(nCoeffs) + z * np.diag(1 / coeffs) @ Q for z in nodes]
+                        vals = [np.linalg.det(K) - 1 for K in kMats]
+                        return np.array(vals)
+
+                    if a is None:
+                        coeffs0 = nodes / m
+                    else:
+                        coeffs0 = a * nodes**b / m
 
-            nCoeffs = M
-            if coll.quad_type in ['LOBATTO', 'RADAU-LEFT']:
-                nCoeffs -= 1
-                Q = Q[1:, 1:]
-                nodes = nodes[1:]
+                    with warnings.catch_warnings():
+                        warnings.simplefilter("ignore")
+                        coeffs = sp.optimize.fsolve(nilpotency, coeffs0, xtol=1e-15)
 
-            def func(coeffs):
-                coeffs = np.asarray(coeffs)
-                kMats = [(1 - z) * np.eye(nCoeffs) + z * np.diag(1 / coeffs) @ Q for z in nodes]
-                vals = [np.linalg.det(K) - 1 for K in kMats]
-                return np.array(vals)
+                # Handle first node equal to zero
+                if quadType in ['LOBATTO', 'RADAU-LEFT']:
+                    coeffs = np.asarray([0.0] + list(coeffs))
+                    nodes = coll.nodes
```

Comment:
Are you adding back the first node here? It's a bit confusing.

Reply:
Yes ... changed to an explicit (although less efficient) solution.